File: src/lib/catalog.ts
import rawPlugins from '../data/plugins.json'
import rawResources from '../data/resources.json'
import type { PluginCatalogEntry } from '../types/plugin'

export const pluginCatalog = [
  ...(rawPlugins as PluginCatalogEntry[]),
  ...(rawResources as PluginCatalogEntry[]),
]

export function getPluginById(pluginId: string): PluginCatalogEntry | undefined {
  return pluginCatalog.find((plugin) => plugin.id === pluginId)
}

File: src-tauri/src/utils/catalog.rs
use crate::models::plugin::PluginCatalogEntry;
use crate::utils::errors::AppError;

const CURATED_CATALOG_JSON: &str = include_str!("../../../src/data/plugins.json");
const IMPORTED_CATALOG_JSON: &str = include_str!("../../../src/data/resources.json");

pub fn load_plugin_catalog() -> Result<Vec<PluginCatalogEntry>, AppError> {
    let mut plugins: Vec<PluginCatalogEntry> = serde_json::from_str(CURATED_CATALOG_JSON)?;
    let mut imported: Vec<PluginCatalogEntry> = serde_json::from_str(IMPORTED_CATALOG_JSON)?;
    plugins.append(&mut imported);
    Ok(plugins)
}
